### DISCUSSION TITLE
Deployment doesn't display package README

### DISCUSSION THREAD

**Discussion by deeeed** [verdaccio/charts#132] [Q&A]
I am using the helm chart and everything is deployed properly but none of my package can show the README in web interface:
![image](https://github.com/verdaccio/charts/assets/4800095/183e474c-29f9-4ca9-ae47-79a931de3bf8)

Is there something I am missing?

I found related issue https://github.com/verdaccio/verdaccio/issues/2469

Is there a way to integrate https://www.npmjs.com/package/verdaccio-local-storage-advance ?

**Comment by deeeed** [comment]
> I am using the helm chart and everything is deployed properly but none of my package can show the README in web interface: ![image](https://user-images.githubusercontent.com/4800095/259609527-183e474c-29f9-4ca9-ae47-79a931de3bf8.png)
> 
> Is there something I am missing?
> 
> I found related issue [verdaccio/verdaccio#2469](https://github.com/verdaccio/verdaccio/issues/2469)
> 
> Is there a way to integrate https://www.npmjs.com/package/verdaccio-local-storage-advance ?

Or maybe have an option to run a script that would install the necessary packages? Creating a custom image seem unproductive and much harder to maintain

**Comment by juanpicado** [comment]
but which package manager are you using? I don't have any issue with README with npm for instance

**Reply by deeeed** [reply]
it seems to be an issue with yarn berry, latest npm, and pnpm. It works when the readme file of package.json is used so I use a prepack / postpack hack:
```
 "scripts": {
    "build": "tsc",
    "generate-docs": "typedoc",
    "test": "echo \"Error: no test specified\" && exit 1",
    "prepack": "node scripts/prepack.js",
    "postpack": "node scripts/postpack.js",
    "version:patch": "yarn version patch && yarn build && yarn generate-docs && yarn npm publish",
    "version:minor": "yarn version minor && yarn build && yarn generate-docs && yarn npm publish",
    "version:major": "yarn version major && yarn build && yarn generate-docs && yarn npm publish"
  },
  ```
  
  ```js prepack
  // fix https://github.com/verdaccio/verdaccio/issues/1905
const path = require('path')
const fs = require('fs')
const packagePath = path.resolve('./package.json')
const readmePath = path.resolve('./README.md')
const package = require(packagePath)
const readme = fs.readFileSync(readmePath, 'utf8')
package.readme = readme
fs.writeFileSync(packagePath, JSON.stringify(package, null, 2))

  ```
  
  ```js postpack
  const path = require('path');
const fs = require('fs');
const packagePath = path.resolve('./package.json');
const package = require(packagePath);

// Remove the readme field
delete package.readme;

// Write the updated package.json back to disk
fs.writeFileSync(packagePath, JSON.stringify(package, null, 2));
  
  ```

**Reply by akasection** [reply]
On my side, even with hacks above, the README.md is still missing. Tested with npm 10.x and pnpm 8.x (latest).

**Reply by juanpicado** [reply]
Could someone create a reproducible steps with clear store installation?

**Reply by akasection** [reply]
So I'm trying to reproduce it via k3d + verdaccio helm, but to no avail. But...

In my existing condition:
- Verdaccio version was 4.7
- Some package called `@namespace/package-name` uploaded using pnpm 8/yarn 2, causing README error.
- Upgraded the Verdaccio to 5.x (latest)
- Bump the `@namepsace/package-name` to newer version and publish 
- The readme is still not read

Hence, I assume, In local, if it's fresh publish, the readme works. If it was "dirtied" before with broken readme, it will never gets corrected, even though the Verdaccio dan pnpm/yarn version are using the latest one.

**Reply by D3strukt0r** [reply]
For anyone that still needs this script, I have here a bit modernised one, executable by [TSX](https://tsx.is/)

`prepack.ts`
```ts
import { appendFileSync, readFileSync, writeFileSync } from 'node:fs';
import { EOL } from 'node:os';
import { resolve } from 'node:path';

const packagePath = resolve('./package.json');
const readmePath = resolve('./README.md');

try {
  const packageContent = readFileSync(packagePath, 'utf8');
  const packageData = JSON.parse(packageContent);

  const readmeContent = readFileSync(readmePath, 'utf8');

  packageData.readme = readmeContent;

  writeFileSync(packagePath, JSON.stringify(packageData, null, 2));
  appendFileSync(packagePath, EOL, 'utf8');

  console.log('✓ README content added to package.json');
} catch (error) {
  console.error('Error during prepack:', error);
  process.exit(1);
}
```

`postpack.ts`
```ts
import { appendFileSync, readFileSync, writeFileSync } from 'node:fs';
import { EOL } from 'node:os';
import { resolve } from 'node:path';

const packagePath = resolve('./package.json');

try {
  const packageContent = readFileSync(packagePath, 'utf8');
  const packageData = JSON.parse(packageContent);

  delete packageData.readme;

  writeFileSync(packagePath, JSON.stringify(packageData, null, 2));
  appendFileSync(packagePath, EOL, 'utf8');

  console.log('✓ README content removed from package.json');
} catch (error) {
  console.error('Error during postpack:', error);
  process.exit(1);
}
```
